File: C++/Medium/WrongAnswer/apply-discount-to-prices_3.cpp
class Solution {
public:
    string discountPrices(string sentence, int discount) {
        
        string result;
        string word;
        for (int i = 0; i < sentence.size(); i++) {
            if (sentence[i] == ' ') {
                if (word.size() > 0 && word[0] == '$') {
                    word.erase(word.begin());
                    double price = stod(word);
                    price = price * (100 - discount) / 100;
                    string priceStr = to_string(price);
                    priceStr = priceStr.substr(0, priceStr.find('.') + 3);
                    word = "$" + priceStr;
                }
                result += word + " ";
                word = "";
            } else {
                word += sentence[i];
            }
        }
        if (word.size() > 0 && word[0] == '$') {
            word.erase(word.begin());
            double price = stod(word);
            price = price * (100 - discount) / 100;
            string priceStr = to_string(price);
            priceStr = priceStr.substr(0, priceStr.find('.') + 3);
            word = "$" + priceStr;
        }
        result += word;
        return result;
    }
};

File: Rust/Medium/RuntimeError/apply-discount-to-prices_3.rs
impl Solution {
    pub fn discount_prices(sentence: String, discount: i32) -> String {
        
        let mut result = String::new();
        for word in sentence.split_whitespace() {
            if word.starts_with("$") {
                let price = word[1..].parse::<f32>().unwrap();
                let discount = (price * discount as f32) / 100.0;
                result.push_str(&format!("${:.2}", price - discount));
            } else {
                result.push_str(word);
            }
            result.push(' ');
        }
        result.pop();
        result
    }
}